src/OrchardCore.Modules/OrchardCore.Contents/Sitemaps/ContentTypesSitemapSourceUpdateHandler.cs

```diff
@@ -40,22 +47,25 @@ public async Task ClearCacheAsync(SitemapCacheContext context)
                         continue;
                     }
 
-                    if (source.IndexAll) {
-                        await _sitemapCacheProvider.ClearSitemapCacheAsync(sitemap.Path);
+                    if (source.IndexAll)
+                    {
+                        sitemap.Identifier = IdGenerator.GenerateId();
                         break;
                     }
                     else if (source.LimitItems && String.Equals(source.LimitedContentType.ContentTypeName, contentTypeName))
                     {
-                        await _sitemapCacheProvider.ClearSitemapCacheAsync(sitemap.Path);
+                        sitemap.Identifier = IdGenerator.GenerateId();
```

Comment:
@jtkech just digging into a regression here.

Nothing is clearing the cache anymore when these event handlers are updated.

Reply:
@deanmarcussen 

Yes i know, we only invalidate it by updating its shared identifier, and the shared identifier is checked on demand in `SitemapController` so that the file is regenerated even it still exists, and in a distributed env the same happens in a given instance A even a sitemap identifier was updated by an instance B.

Then there is the `PurgeAll()` command that would need to be distributed, but not yet done

Then one idea would be to also clean them by checking all sitemap identifiers (in the shared sitemap document) e.g. through a background task. I recently worked on WF atomicity, next step is to start working on atomic / exclusive background tasks, hmm maybe an exclusive background task would be a good candidate to clean cached files.